remote-method/index.js

```diff
@@ -0,0 +1,285 @@
+'use strict';
+var yeoman = require('yeoman-generator');
+var chalk = require('chalk');
+
+var actions = require('../lib/actions');
+var helpers = require('../lib/helpers');
+var validateName = helpers.validateName;
+var typeChoices =helpers.getTypeChoices;
+var accepts = [];
+var returns = [];
+var https = [];
+
+module.exports = yeoman.generators.Base.extend({
+  // This generator does not track file changes via yeoman,
+  // as loopback-workspace is editing (modifying) files when
+  // saving project changes.
+  constructor: function() {
+    yeoman.generators.Base.apply(this, arguments);
+
+    this.argument('modelName', {
+      desc: 'Name of the model',
+      required: false,
+      type: String
+    });
+  },
+
+  help: function() {
+    return helpers.customHelp(this);
+  },
+
+  loadProject: actions.loadProject,
+
+  loadModels: actions.loadModels,
+
+  askForModel: function() {
+    if (this.modelName) {
+      this.modelDefinition = this.projectModels.filter(function(m) {
+        return m.name === this.modelName;
+      }.bind(this))[0];
+
+      if(!this.modelDefinition) {
+        var msg = 'Model not found: ' + this.modelName +
+                  ' Please choose from Model List:';
+        this.log(chalk.red(msg));
+      }
+    }
+
+    if (!this.modelDefinition) {
+      var done = this.async();
+      var prompts = [
+        {
+          name: 'model',
+          message: 'Select the model:',
+          type: 'list',
+          choices: this.modelNames
+        }
+      ];
+      this.prompt(prompts, function(answers) {
+        this.modelName = answers.model;
+        done();
+      }.bind(this));
+    }
+  },
+
+  findModelDefinition: function() {
+    this.modelDefinition = this.projectModels.filter(function(m) {
+        return m.name === this.modelName;
+    }.bind(this))[0];
+
+    if (!this.modelDefinition) {
+        var msg = 'Model not found: ' + this.modelName;
+        this.log(chalk.red(msg));
+        this.async()(new Error(msg));
```

Comment:
hmm...actually I use `this.async()` to keep it consistent with other files, eg:
https://github.com/strongloop/generator-loopback/blob/master/relation/index.js#L54
https://github.com/strongloop/generator-loopback/blob/master/property/index.js#L56
And I tried `throw new Error(msg)` it has the same performance, so that means somewhere in yeoman catches that error. I will check yeoman doc to confirm that. 
And could you please explain it more like why `throw` is simpler than `this.async` ? Both of them work me but I am a little confused with their difference (in terms of speed, etc), thank you so much.